TradeItIosTicketSDK2/TradeItPortfolioAccountsTableViewManager.swift

```diff
@@ -25,74 +20,132 @@ class TradeItPortfolioAccountsTableViewManager: NSObject, UITableViewDelegate, U
-        self.linkedBrokersInError = linkedBrokersInError
+    func updateAccounts(withLinkedBrokers linkedBrokers: [TradeItLinkedBroker]) {
+        self.linkedBrokerSectionPresenters = linkedBrokers.map({ linkedBroker in
+            return LinkedBrokerSectionPresenter(linkedBroker: linkedBroker)
+        })
         self.accountsTable?.reloadData()
-
-        if accounts.count > 0 {
-            let selectedAccount = selectedAccount ?? self.accounts[0]
-            let selectedAccountIndex = self.accounts.index(where: {$0.accountNumber == selectedAccount.accountNumber && $0.brokerName == selectedAccount.brokerName}) ?? 0
-            let selectedIndexPath = IndexPath(row: selectedAccountIndex, section: 0)
-            self.accountsTable?.selectRow(at: selectedIndexPath, animated: true, scrollPosition: .top)
-            self.delegate?.linkedBrokerAccountWasSelected(selectedAccount: selectedAccount)
-        } else if self.linkedBrokersInError.count > 0 {
-            let selectedIndexPath = IndexPath(row: 0, section: 0)
-            self.accountsTable?.selectRow(at: selectedIndexPath, animated: true, scrollPosition: .top)
-            self.delegate?.linkedBrokerInErrorWasSelected(selectedBrokerInError: self.linkedBrokersInError[0])
-        }
     }
 
     // MARK: UITableViewDelegate
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if indexPath.row < self.accounts.count {
-            let selectedAccount = self.accounts[indexPath.row]
-            self.delegate?.linkedBrokerAccountWasSelected(selectedAccount: selectedAccount)
+        guard let account = self.linkedBrokerSectionPresenters[safe: indexPath.section - 1]?.accountFor(row: indexPath.row) else { return }
+        self.delegate?.linkedBrokerAccountWasSelected(selectedAccount: account)
+    }
+
+    // MARK: UITableViewDataSource
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1 + self.linkedBrokerSectionPresenters.count
+    }
+
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if (section == 0) {
+            return "Total Value"
         } else {
-            let linkedBrokerInError = self.linkedBrokersInError[indexPath.row - self.accounts.count]
-            self.delegate?.linkedBrokerInErrorWasSelected(selectedBrokerInError: linkedBrokerInError)
+            return self.linkedBrokerSectionPresenters[safe: section - 1]?.linkedBroker.brokerName
         }
     }
 
-    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 30.0
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if (section == 0) {
+            return 1
+        } else {
+            guard let linkedBrokerSectionPresenter = self.linkedBrokerSectionPresenters[safe: section - 1] else { return 0 }
+            return linkedBrokerSectionPresenter.numberOfRows()
```

Comment:
This is incorrect if the broker has an error. In that case it should have +1 row.